### ISSUE TITLE
[Bug]: gauche CANNOT LINK EXECUTABLE "gosh": cannot locate symbol "AO_pt_lock"

### ISSUE THREAD

**Issue by dauuricus** [termux/termux-packages#8329]
### Problem description

CANNOT LINK EXECUTABLE "gosh": cannot locate symbol "AO_pt_lock" referenced by "/data/data/com.termux/files/usr/lib/gauche-0.97/0.9.10/arm-unknown-linux-androideabi/libgauche-0.97.so.0.10"...

### What steps will reproduce the bug?

command 'gosh' :gauche 

### What is the expected behavior?

i want use gauche with embeded mbedtls (3.0.0) for https.
it's nice.

### System information

termux-info:

```
Application version:
0.108
Packages CPU architecture:
arm
Subscribed repositories:
# sources.list
deb https://deb.kcubeterm.me/termux-main/ stable main
# sources.list.d/cctools.list
deb https://cctools.info termux cctools
# sources.list.d/pointless.list
deb https://its-pointless.github.io/files/24 termux extras
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.org/apt/termux-x11 x11 main                              
# game-repo (sources.list.d/game.list)
deb https://grimler.se/game-packages-24 games stable
# science-repo (sources.list.d/science.list)
deb https://grimler.se/science-packages-24 science stable                            Updatable packages:
gnutls/stable 3.6.16-1 arm [upgradable from: 3.6.16]
libgnutls/stable 3.6.16-1 arm [upgradable from: 3.6.16]
util-linux/stable 2.37.2-1 arm [upgradable from: 2.37.2]
Android version:9
Kernel build information:
Linux localhost 4.4.146+ #1 SMP PREEMPT Thu Nov 25 16:54:56 UTC 2021 armv8l Android  Device manufacturer:
Amazon                                                                               
Device model:                                                                        KFMAWI
```

**Comment by Grimler91** [comment]
Can confirm the issue on arm. Aarch64 works though

**Comment by xtkoba** [comment]
I will look into this.

**Comment by xtkoba** [comment]
Another issue arises for arm:

```
"/home/builder/.termux-build/gauche/src/src/list.c", line 798 (ScmExtendedPairDescriptor *Scm__GetExtendedPairDescriptor(ScmObj)): Assertion failed: (z->hiddenTag&0x7) == 0x7
```

As I have no idea how to fix this, I have decided to disable arm build for now. But feel free to try to fix this for yourself, as well as the issue for thread support. Thanks!

**Comment by xtkoba** [comment]
The aforementioned assertion failure is resolved by PR #8403. Thanks!
